pkg/actions/irsa/update.go

```diff
@@ -27,12 +28,36 @@ func (a *Manager) UpdateIAMServiceAccounts(iamServiceAccounts []*api.ClusterIAMS
 	for _, iamServiceAccount := range iamServiceAccounts {
 		stackName := makeIAMServiceAccountStackName(a.clusterName, iamServiceAccount.Namespace, iamServiceAccount.Name)
 
-		if _, ok := existingIAMStacksMap[stackName]; !ok {
+		stack, ok := existingIAMStacksMap[stackName]
+		if !ok {
 			logger.Info("cannot update IAMServiceAccount %s/%s as it does not exist", iamServiceAccount.Namespace, iamServiceAccount.Name)
 			nonExistingSAs = append(nonExistingSAs, fmt.Sprintf("%s/%s", iamServiceAccount.Namespace, iamServiceAccount.Name))
 			continue
 		}
 
+		var roleName string
+		for _, o := range stack.Outputs {
+			if aws.StringValue(o.OutputKey) != outputs.IAMServiceAccountRoleName {
+				continue
+			}
+			roleArn, err := arn.Parse(aws.StringValue(o.OutputValue))
+			if err != nil {
+				return fmt.Errorf("failed to parse role arn %q: %w", aws.StringValue(o.OutputValue), err)
+			}
+			start := strings.IndexRune(roleArn.Resource, '/')
+			if start == -1 {
+				return fmt.Errorf("failed to parse resource: %s", roleArn.Resource)
+			}
+			roleName = roleArn.Resource[start+1:]
+			logger.Info("found role name %q for service account role", roleName)
+			break
+		}
+
+		if roleName == "" {
+			return fmt.Errorf("failed to find role name for Role1 for service account: %s", iamServiceAccount.Name)
+		}
```

Comment:
worth moving to a help func? Making the main func quite noisy

Reply:
done